storage/tests/c/flash.h

```diff
@@ -24,7 +24,19 @@
 #include <stdlib.h>
 #include "secbool.h"
 
-#define FLASH_BYTE_ACCESS 1
+#define FLASH_BLOCK_SIZE (sizeof(uint32_t) * FLASH_BLOCK_WORDS)
+
+typedef uint32_t flash_block_t[FLASH_BLOCK_WORDS];
+
+#if FLASH_BLOCK_WORDS == 1
+#define FLASH_ALIGN(X) (((X) + 3) & ~3)
+#define FLASH_IS_ALIGNED(X) (((X)&3) == 0)
+#elif FLASH_BLOCK_WORDS == 4
+#define FLASH_ALIGN(X) (((X) + 0xF) & ~0xF)
+#define FLASH_IS_ALIGNED(X) (((X)&0xF) == 0)
+#else
+#error Unsupported number of FLASH_BLOCK_WORDS.
+#endif
```

Comment:
Darn, I though I put these in flash_common.h, so that we don't have to redefine them elsewhere (core/embed/trezorhal/flash.h, legacy/flash.h). Also some of the declarations below could probably also be moved to flash_common.h, (some going under #ifdef FLASH_BIT_ACCESS).

Reply:
Actually the functions that are declared below should go into some internal header file, because they are not meant to be used directly anymore now that we have the `flash_area_*` functions. They are only meant for the implementation of the `flash_area_*` functions. But this particular refactoring job can be handled in a separate issue.

Reply:
reverted in https://github.com/trezor/trezor-firmware/commit/87ca3a5654dd208defdaf5a63216d8ea748194e1

moved the flags to build scripts from header file to make it compile.

note that `flash_common.h` and related stuff get refactored in https://github.com/trezor/trezor-firmware/pull/3370 , specifically in commit https://github.com/trezor/trezor-firmware/pull/3370/commits/757fed5ef4fa3c3c906a607b9096e0427f32ed08 , which includes creating the internal header file called `flash_ll.h`. `flash_common.h` is changed into `flash_area.h` and these new definitions will probably be moved to `flash_ll`, so the placement of these macros here is temporary anyway.